### PR TITLE
Docs: Enhance Flink pages

### PR DESCRIPTION
**Pull request by manuzhang**
1. Fix internal links
2. Remove period in title
3. Fix numbered list with code blocks
4. Add identifier fields as an approach for upsert mode

### CODE REVIEW

File: docs/docs/flink-writes.md [hunk 1]
```diff
@@ -59,20 +59,20 @@ Iceberg supports `UPSERT` based on the primary key when writing data into v2 tab
...
 1. Enable the `UPSERT` mode as table-level property `write.upsert.enabled`. Here is an example SQL statement to set the table property when creating a table. It would be applied for all write paths to this table (batch or streaming) unless overwritten by write options as described later.
 
-```sql
-CREATE TABLE `hive_catalog`.`default`.`sample` (
-    `id` INT COMMENT 'unique id',
-    `data` STRING NOT NULL,
-    PRIMARY KEY(`id`) NOT ENFORCED
-) with ('format-version'='2', 'write.upsert.enabled'='true');
-```
+    ```sql
```
Comment (nastra) [inline docs/docs/flink-writes.md:62]: I don't think this should have indentation

File: docs/docs/flink-writes.md [hunk 2]
```diff
@@ -59,20 +59,20 @@ Iceberg supports `UPSERT` based on the primary key when writing data into v2 tab
...
+    ```
 
-2. Enabling `UPSERT` mode using `upsert-enabled` in the [write options](#write-options) provides more flexibility than a table level config. Note that you still need to use v2 table format and specify the primary key when creating the table.
+2. Enabling `UPSERT` mode using `upsert-enabled` in the [write options](#write-options) provides more flexibility than a table level config. Note that you still need to use v2 table format and specify the [primary key](../flink-ddl.md/#primary-key) or [identifier fields](../../spec.md#identifier-field-ids) when creating the table.
 
-```sql
-INSERT INTO tableName /*+ OPTIONS('upsert-enabled'='true') */
-...
-```
+    ```sql
```
Comment (nastra) [inline docs/docs/flink-writes.md:72]: same as above

**Comment by manuzhang** [comment]
@nastra This is how the numbered list with code blocks look like without indentation

CleanShot 2024-03-11 at 17 30 00@2x

**Comment by nastra** [comment]
@manuzhang is it then correctly intended with your changes? Can you post a screenshot please?

**Comment by manuzhang** [comment]
Here it is.

CleanShot 2024-03-11 at 18 28 14@2x

**Review by nastra — APPROVED** [review]

**Pull request merged**